flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java

```diff
@@ -93,6 +93,12 @@
 	/** User supplied deserialization schema to convert Kinesis byte messages to Flink objects. */
 	private final KinesisDeserializationSchema<T> deserializer;
 
+	/**
+	 * The function that determines which subtask a shard should be assigned to.
+	 */
+	// TODO: instead of the property, use a factory method that would allow subclass to access source context?
```

Comment:
createFn(...) that will allow the function to be created with access to runtime context (like the number of subtasks), and then change the fn signature to only take shard metadata as parameter. Subclasses can override createFn, instead of having the property.

Reply:
Please see my comments below.